examples/model_selection/plot_tuned_decision_threshold.py

```diff
@@ -0,0 +1,377 @@
+# is used to make decisions. Besides, the metric used to tune the decision threshold
+# should be chosen carefully. Here, we used the balanced accuracy but it might not be
+# the most appropriate metric for the problem at hand. The choice of the "right" metric
+# is usually problem-dependent and might require some domain knowledge. Refer to the
+# example entitled,
+# :ref:`sphx_glr_auto_examples_model_selection_plot_cost_sensitive_learning.py`,
+# for more details.
+#
+# Tuning the decision threshold under constraint
+# ----------------------------------------------
+#
+# In some cases, we do not want only to maximize a specific metric but instead maximize
+# a metric while satisfying a constraint on another metric. In the current example, we
+# could imagine that the decision of our predictive model will be reviewed by a medical
+# doctor. In this case, this doctor will only accept a ratio of false positive.
+# Therefore, we are interesting at maximizing the true positive rate while having a
+# a false positive rate lower than a given threshold.
+#
+# The :class:`~sklearn.model_selection.TunedThresholdClassifier` allows to tune the
+# decision threshold with such specification. We should how to proceed using a single
+# cross-validation split to display the Receiver Operating Characteristic (ROC) curves
+# to get intuition on the problem.
+#
+# First, we split the data into a training and testing set.
+
+# %%
+from sklearn.model_selection import train_test_split
+
+data_train, data_test, target_train, target_test = train_test_split(
+    data, target, random_state=42
+)
+
+# %%
+# Now, we will train both the vanilla and tuned model on the training set. We recall
+# that the tuned model is internally maximizing the balanced accuracy for the moment.
+model.fit(data_train, target_train)
+tuned_model.fit(data_train, target_train)
+
+# %%
+# To show the benefit on optimizing a metric under constraint, we will evaluate the
+# models using the ROC curves statistics: the true positive rate (TPR) and the false
+# positive rate (FPR).
+#
+# The FPR is not defined in scikit-learn and we define it below:
+from sklearn.metrics import confusion_matrix, make_scorer, recall_score
+
+
+def fpr_score(y, y_pred, neg_label, pos_label):
+    cm = confusion_matrix(y, y_pred, labels=[neg_label, pos_label])
+    tn, fp, _, _ = cm.ravel()
+    tnr = tn / (tn + fp)
+    return 1 - tnr
+
+
+tpr_score = recall_score  # TPR and recall are the same metric
+scoring = {
+    "fpr": make_scorer(fpr_score, neg_label=neg_label, pos_label=pos_label),
+    "tpr": make_scorer(tpr_score, pos_label=pos_label),
+}
+
+# %%
+# Now, we plot the ROC curve of both models and the FPR and TPR statistics for the
+# decision thresholds of both models.
+from sklearn.metrics import RocCurveDisplay
+
+disp = RocCurveDisplay.from_estimator(
+    model, data_test, target_test, name="Vanilla model", linestyle="--", alpha=0.5
+)
+RocCurveDisplay.from_estimator(
```

Comment:
I would add the roc curve display earlier, since to me the most natural way to think about picking the threshold is picking a point on the ROC curve. Usually when I teach this, I show the ROC curve and then the point that's picked by the default threshold and I'm like "well that doesn't seem very special why don't we pick a better one?".